libs/application/templates/meta-application/src/fields/DataTable/index.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { FC, useMemo } from 'react'
+import {
+  Application,
+  formatText,
+  RepeaterProps,
+} from '@island.is/application/core'
+import {
+  Box,
+  ButtonDeprecated as Button,
+  Text,
+} from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import { Table } from '@island.is/shared/table'
+
+import { m } from '../../forms/messages'
+import isEmpty from 'lodash/isEmpty'
+
+interface DataTableProps {
+  application: Application
+}
+
+interface Data {
+  name: string
+  publisher: string
+  download: string
+  upload: string
+}
+
+const DataTable: FC<DataTableProps> = ({ application }) => {
+  const { formatMessage } = useLocale()
+  const isDraft = application.state === 'draft'
+
+  const emptyMessage = formatText(
+    isDraft ? m.dataEmptyDraft : m.dataEmpty,
+    application,
+    formatMessage,
+  )
+
+  const data = useMemo(
+    () => (application as { answers?: { data?: Data[] } }).answers?.data ?? [],
```

Comment:
I do not understand this useMemo, what are you trying to achieve?

could you do: `const data = application.answers.data ?? []` instead ?

Reply:
React-table requires the data to be memoized https://react-table.tanstack.com/docs/api/useTable

Reply:
ok makes sense